pkg/controllers/user/alert/configsyncer/configsyncer.go

```diff
@@ -556,7 +556,7 @@ func (d *ConfigSyncer) addRecipients(notifiers []*v3.Notifier, receiver *alertco
 					slack.Channel = r.Recipient
 				}
 
-				if notifier.Spec.SlackConfig.HTTPClientConfig != nil {
+				if notifier.Spec.SlackConfig.HTTPClientConfig != nil && notifier.Spec.SlackConfig.HTTPClientConfig.ProxyURL != "" {
```

Comment:
Given that this check happens many times I think we can have a utility function for it.
e.g.:
```
func IsHTTPClientConfigSet(cfg *v3.HTTPClientConfig) bool{}
```